Add Ruff linter with permissive config

mikaelarguedas opened the pull request:
Follow-up of https://github.com/XavB64/lowtrip/pull/56

This currently ignores a lot of violations in the code. need to get authors/maintainers' feedback on:
- is suggested style satisfactory
- strategy for ignored rules

mikaelarguedas commented on backend/pyproject.toml:
```diff
@@ -0,0 +1,157 @@
+[tool.isort]
+py_version=312
```
here we need to put the python version actually used by the project

XavB64 replied:
Hi, I believe it's 3.9, so I just change 312 in 3.9 ?

mikaelarguedas replied:
indeed: 312 -> 39

done in https://github.com/XavB64/lowtrip/pull/69/commits/c90d4aff98d74c0067eb04579bda31f5cdc4ee7d

mikaelarguedas commented on backend/pyproject.toml:
```diff
@@ -0,0 +1,157 @@
+# Same as Black.
+indent-width = 4
+
+target-version = "py312"
```
here we need to put the python version actually used by the project

mikaelarguedas commented on backend/pyproject.toml:
```diff
@@ -0,0 +1,157 @@
+
+[tool.ruff.lint]
+preview = true
+ignore = [
```
here is the list of ruff violations currently ignored in this PR,
Ideally these would be addressed in follow-up PRs instead of ignored long term

mikaelarguedas commented on backend/transport.py:
```diff
@@ -125,49 +125,46 @@ def train_to_gdf(
+        print("Rescaling factor", train_dist / gdf["path_length"].sum())
+        gdf["path_length"] *= train_dist / gdf["path_length"].sum()
+        # Compute emissions : EF * length
+        gdf["EF_tot"] /= 1000.0  # Conversion in in kg
```
```suggestion
        gdf["EF_tot"] /= 1000.0  # Conversion in kg
```

mikaelarguedas commented on backend/utils.py:
```diff
@@ -18,32 +18,37 @@
 ###################
 ###### Utils ######
 ###################
+from http import HTTPStatus
```
here we import a new module from the Python standard library to use the Enum for http statuses instead of hardcoding the return codes

mikaelarguedas commented on backend/utils.py:
```diff
@@ -405,18 +416,19 @@ def find_bicycle(tag1, tag2):
         print("Bicycle length", round(route_dist, 1))
     else:
-        geom_route, route, route_idst = None, False, None
+        geom_route, route, route_dist = None, False, None
```
this fixes a typo that likely fixes a bug

clarani reviewed (approved):
LGTM :white_check_mark:

mikaelarguedas commented:
This has been rebased and fixed formatting issues introduced in https://github.com/XavB64/lowtrip/commit/128b71397578f2801a8df87d2623fac0320fe0ab

@clarani do you know what Python version is used for deployment ? this way https://github.com/XavB64/lowtrip/pull/69#discussion_r1615030483 and https://github.com/XavB64/lowtrip/pull/69#discussion_r1615030736 can be adjusted accordingly before merging

Thanks !

XavB64 commented:
I changed the python version to 3.11 as its 3.11.7 in the deployment on Render, but a check has failed

mikaelarguedas commented:
> I changed the python version to 3.11 as its 3.11.7 in the deployment on Render, but a check has failed... Maybe caused by the "strict" parameter added in the code, is it necessary ?

Yes that's one of the rules I ignored originally the time we converge on a first permissive set of rules to not make too many changes in the code.

This rule didnt apply to python3.9 but is active for python 3.11.

The details of why using zip without setting the strict is dangerous is detailed here: 
https://docs.astral.sh/ruff/rules/zip-without-explicit-strict/

For now I added back this rule to the list of ignored rules so this can move forward: https://github.com/XavB64/lowtrip/pull/69/commits/146de02a37d4e1cee82bbde67dbcce3fbdb93733

---

The full list of rules ignored for now is here:  https://github.com/XavB64/lowtrip/blob/146de02a37d4e1cee82bbde67dbcce3fbdb93733/backend/pyproject.toml#L79-L136

On a follow-up we could see which ones should be reactivated and fixed in the short term to improve code reliability
